pkg/api/api.go

```diff
@@ -607,10 +608,10 @@ func (hs *HTTPServer) registerRoutes() {
 	r.Get("/avatar/:hash", requestmeta.SetSLOGroup(requestmeta.SLOGroupHighSlow), hs.AvatarCacheServer.Handler)
 
 	// Snapshots
-	r.Post("/api/snapshots/", reqSnapshotPublicModeOrSignedIn, hs.getCreatedSnapshotHandler())
+	r.Post("/api/snapshots/", reqSnapshotPublicModeOrCreate, hs.getCreatedSnapshotHandler())
 	r.Get("/api/snapshot/shared-options/", reqSignedIn, hs.GetSharingOptions)
```

Comment:
Feels a bit weird to protect creation and deletion with RBAC, but not listing.

Reply:
I've added a new role but this is not the endpoint to list snapshots, it is above https://github.com/grafana/grafana/pull/96126/files#diff-67a6e36014be865053c43d38b2091bd2d3f17a91b117d97c0f5244cd7d71127dR516

Reply:
> I discovered that listing snapshots has a mechanism in the [database layer](https://github.com/grafana/grafana/blob/a3fb2109791483125ef9947a332f0b68e7babd87/pkg/services/dashboardsnapshots/database/database.go#L134).

Oh, interesting. I think it still makes sense to keep the permission, but I can imagine cases where people might be confused about not being able to list all snapshots even though they have snapshot reader permission. But let's deal with it if/when we get complaints about it.